### PR TITLE
[438] - [FE] Integrate Fetch and Filter Notifications functionality according to a selected project

### PR DESCRIPTION
**Pull request by rogeliojohnoliverio**
## Backend
- https://github.com/abduljalilpalala/ps-slackana/pull/138
## Asana Link
- https://app.asana.com/0/1203011167276287/1203247946500438/f

## Definition of Done
- [x] User can now see all notifications depending on the selected project
- [x] User can filter notifications according to a selected project
- [x] User can filter notification according to type such as all or unread

## Notes
- Please run `php artisan migrate` before running the server as I have minor fix on the backend

## Pre-condition
- Login on the app
- Click the notification bell
- Click on see all notifications

## Expected Output
- Should be able see and filter notifications according to a selected project or filter unreads only or all

## Screenshots/Recordings

https://user-images.githubusercontent.com/110364637/199702923-8911fb8b-9d04-462b-b714-300353a3c3a5.mp4

### CODE REVIEW

File: client/src/components/molecules/NotificationList/index.tsx [hunk 1]
```diff
@@ -3,14 +3,19 @@ import React, { FC } from 'react'
 import { GitHub } from 'react-feather'
 import { BsGithub } from 'react-icons/bs'
 import { FaRegUser } from 'react-icons/fa'
+import ImageSkeleton from '~/components/atoms/Skeletons/ImageSkeleton'
+import LineSkeleton from '~/components/atoms/Skeletons/LineSkeleton'
+import { useAppSelector } from '~/hooks/reduxSelector'
```
Comment (abduljalilpalala) [inline client/src/components/molecules/NotificationList/index.tsx:9]: Separate Imports.

File: client/src/components/molecules/NotificationList/index.tsx [hunk 2]
```diff
@@ -3,14 +3,19 @@ import React, { FC } from 'react'
...
 
 import { Notification } from '~/shared/interfaces'
```
Comment (abduljalilpalala) [inline client/src/components/molecules/NotificationList/index.tsx]: Separate Imports.

```suggestion
import { GitHub } from 'react-feather'
import { BsGithub } from 'react-icons/bs'
import { FaRegUser } from 'react-icons/fa'

import ImageSkeleton from '~/components/atoms/Skeletons/ImageSkeleton'
import LineSkeleton from '~/components/atoms/Skeletons/LineSkeleton'
import { useAppSelector } from '~/hooks/reduxSelector'
import { Notification } from '~/shared/interfaces'
```

File: client/src/hooks/notificationMethods.ts [hunk 3]
```diff
@@ -4,17 +4,41 @@ import { axios } from '~/shared/lib/axios'
...
+  }
+  const useGetNotificationsTableOnUpdate = async (project_id: number) => {
+    const response = await axios.get(`/api/notification?project=${project_id}&type=all`)
+    if (currentProjectID === project_id) {
+      setNotificationsTable(response.data)
+    }
+  }
   const useGetNotificationsOnMount = async () => {
     const response = await axios.get(`/api/notification`)
     const hasUnreadNotifications = response.data.some((notification: any) => !notification.is_seen)
```
Comment (abduljalilpalala) [inline client/src/hooks/notificationMethods.ts]: Add types instead of `any`

File: client/src/pages/notifications/projects/[id].tsx [hunk 4]
```diff
@@ -34,12 +60,13 @@ const Notifications: NextPage = (): JSX.Element => {
...
             <ul className="flex flex-col space-y-1 text-sm">
-              {teams.map(({ id, name }) => (
-                <li key={id}>
-                  <Link href={`/notifications/projects/${id}`}>
-                    <a
-                      href="#"
-                      className={`
+              {!isSidebarLoading
+                ? sidebarProject?.map(({ id, title }: any) => (
+                    <li key={id}>
```
Comment (abduljalilpalala) [inline client/src/pages/notifications/projects/[id].tsx]: Add types instead of `any`

File: client/src/pages/notifications/projects/[id].tsx [hunk 5]
```diff
@@ -65,24 +93,30 @@ const Notifications: NextPage = (): JSX.Element => {
...
-                {teams.map(({ id, name }) => (
-                  <option value={name} key={id} className="line-clamp-1">
-                    {name}
+              <select
+                onChange={handleSelectProject}
+                css={globals.form_control}
+                className="w-full font-semibold"
+              >
+                {sidebarProject?.map(({ id, title }: any) => (
+                  <option value={id} key={id} className="line-clamp-1">
```
Comment (abduljalilpalala) [inline client/src/pages/notifications/projects/[id].tsx]: Add types instead of `any`

**Review by abduljalilpalala — COMMENTED** [review]
Please check comments.

File: client/src/components/molecules/NotificationList/index.tsx [hunk 6]
```diff
@@ -23,55 +28,89 @@ const NotificationList: FC<Props> = (props): JSX.Element => {
...
+          <LineSkeleton className="h-[10px] w-6" />
+          <LineSkeleton className="h-[10px] w-96" />
+        </div>
+        <span className="shrink-0 text-sm font-light text-slate-500">
+          <LineSkeleton className=" h-[10px] w-40" />
+        </span>
+      </td>
+    </tr>
+  )
+
```
Comment (ww-jermaine) [inline client/src/components/molecules/NotificationList/index.tsx]:
```js
const tableLoading: FC = (): JSX.Element => {
  return (
    <tr key={Math.random()} className={`group flex-1 shrink-0  text-sm `}>
      <td className="flex flex-1 items-center px-6 md:justify-between">
        <div className="flex flex-wrap items-center space-x-2 py-2 text-slate-600">
          <ImageSkeleton className="max-h-[30px] max-w-[30px] rounded-md" />
          <LineSkeleton className="h-[10px] w-40" />
          <LineSkeleton className="h-[10px] w-6" />
          <LineSkeleton className="h-[10px] w-96" />
        </div>
        <span className="shrink-0 text-sm font-light text-slate-500">
          <LineSkeleton className=" h-[10px] w-40" />
        </span>
      </td>
    </tr>
  )
}
```

**Review by ww-jermaine — APPROVED** [review]

**Review by jsvelte — APPROVED** [review]
# Approved 💯

**Pull request merged**